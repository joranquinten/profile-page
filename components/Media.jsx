import React from 'react';
import renderHTML from 'react-render-html';
import styled from 'styled-components';

import { Tag, Card } from 'antd';

import { sorterByProperty } from '../common/helpers';
import { media } from '../data/nl';

const typeColorMap = {
  website: 'volcano',
  podcast: 'gold',
  newsletter: 'purple',
};

const StyledMedia = styled.div`
  .ant-tag {
    margin: 0.4rem !important;
    display: inline-block;
  }
`;

const Media = () => (
  <StyledMedia>
    <Card title="Media">
      {renderHTML(media.description)}
      {media.list.sort(sorterByProperty('name')).map(item => (
        <Tag key={item.name} color={typeColorMap[item.type]}>
          {item.name}
        </Tag>
      ))}
    </Card>
  </StyledMedia>
);

export default Media;
