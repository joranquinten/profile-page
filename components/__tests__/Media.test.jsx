import { shallow } from 'enzyme';

import Media from '../Media';

jest.mock('../../data/nl.js');

describe('<Media />', () => {
  it('should render without crashing', () => {
    expect(shallow(<Media />)).toMatchSnapshot();
  });
});
