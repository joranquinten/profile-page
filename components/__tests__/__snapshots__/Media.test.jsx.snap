// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Media /> should render without crashing 1`] = `
<Media__StyledMedia>
  <Card
    title="Media"
  >
    <p
      key="0"
    >
      mediaDescription
    </p>
    <Tag
      closable={false}
      color="gold"
      key="mediaName1"
    >
      mediaName1
    </Tag>
    <Tag
      closable={false}
      color="gold"
      key="mediaName2"
    >
      mediaName2
    </Tag>
    <Tag
      closable={false}
      color="purple"
      key="mediaName3"
    >
      mediaName3
    </Tag>
    <Tag
      closable={false}
      color="volcano"
      key="mediaName4"
    >
      mediaName4
    </Tag>
  </Card>
</Media__StyledMedia>
`;
